embassy-rp/src/spi.rs

```diff
@@ -78,26 +87,48 @@ impl<'d, T: Instance> Spi<'d, T> {
 
     pub fn new_txonly(
         inner: impl Peripheral<P = T> + 'd,
+        tx_dma: Option<PeripheralRef<'d, AnyChannel>>,
```

Comment:
these should be `impl Peripheral<P = impl Channel> + 'd` so that you can't pass None. We don't want to allow creating async SPIs without DMA.